gems/smithy/lib/smithy/generators.rb

```diff
@@ -16,5 +15,15 @@ def self.generate(plan)
         raise 'Not yet implemented'
       end
     end
+
+    def self.source(plan)
+      case plan.type
+      when :schema then Schema.new(plan).source
+      when :client then Client.new(plan).source
+      when :server then raise 'Will not be implemented'
```

Comment:
If we will not implement it, why do we have this?

Reply:
This is for source only generation. I don't think we will write out an entire server implementation as a string. That may change depending on how that's implemented (maybe it's just a shim to include on rails app). The other condition is for unknown option.